### PR TITLE
purism-ectool: init at 0.3.5

### PR DESCRIPTION
**Pull request by avieth**
###### Motivation for this change

Adds the os-specific/linux Purism EC tool

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/purism/ectool.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, fetchgit, rustPlatform, udev, pkg-config }:
+let
+
+  gitsrc = fetchgit {
+      url = "https://source.puri.sm/firmware/librem-ec.git";
+      rev = "05d9990d7500e1cf3d09844407f77cb64c049a12";
+      sha256 = "145x8v5zmc9vp9hci0hf3lczyi1bf8hj4bawyy5vgy8bpvdsjgk7";
+    };
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/purism/ectool.nix:8]:
```suggestion
  gitsrc = fetchgit {
    url = "https://source.puri.sm/firmware/librem-ec.git";
    rev = "05d9990d7500e1cf3d09844407f77cb64c049a12";
    sha256 = "145x8v5zmc9vp9hci0hf3lczyi1bf8hj4bawyy5vgy8bpvdsjgk7";
  };
```
also this should use fetchFromGitLab.

File: pkgs/os-specific/linux/purism/ectool.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+in
+rustPlatform.buildRustPackage rec {
+  pname = "purism-ectool";
+  version = "0.3.5";
+
+  src = gitsrc + /tool;
+
+  cargoSha256 = "sha256:16ahi0qpwfk7vc2wymishg19rbyadv4sk1hp0qzsjj0m3s9ws3yh";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/purism/ectool.nix:17]:
```suggestion
  cargoSha256 = "16ahi0qpwfk7vc2wymishg19rbyadv4sk1hp0qzsjj0m3s9ws3yh";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -326,6 +326,8 @@ with pkgs;
 
   deviceTree = callPackage ../os-specific/linux/device-tree {};
 
+  purism-ec-tool = callPackage ../os-specific/linux/purism/ectool.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:329]:
```suggestion
  purism-ec-tool = callPackage ../os-specific/linux/purism/ectool {};
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134999` at ea4302c8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>purism-ectool</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/purism/ectool/default.nix:27:5:
```
   |
27 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
@avieth do you wish this to get merged? There are merge conflicts now but it seems good otherwise.

**Comment by avieth** [comment]
> @avieth do you wish this to get merged? There are merge conflicts now but it seems good otherwise.

Turns out Purism grifts faulty hardware, therefore I don't want to be a maintainer on this software. Perhaps we should abandon this.

**Pull request closed without merging**